Window knocking

Bod9001 opened the pull request:
### Purpose
Fixes #2469
and some auto format
### Notes:
 I've had to Change CmdRequestMeleeAttack  to Except the intent
so There might be some funny Behaviours caused because of this

This could be implemented in a better way but no one said anything about it on Discord so I went ahead with it

### Please make sure you have followed the self checks below before submitting a PR:

Tested in editor
Will test in build, editor Multiplayer /
Renamed functions to appropriate names

Bod9001 committed:
Window knocking and auto format

Bod9001 committed:
Fixed Stray actions

Bod9001 committed:
Reduced loudness of muffled sounds

chairbender reviewed (changes requested):
@Bod9001 this isn't quite the right way to do this as far as the interaction logic is concerned.  This is a tile interaction, so it should use the Tile Interaction system: https://github.com/unitystation/unitystation/wiki/Implementing-Interactions#how-can-i-create-tile-interactions

This ticket shouldn't require changing the CmdRequestMeleeAttack or TilemapDamage. All that should be needed is to 
  * modify Meleeable, moving LayerType.Windows into harmIntentOnlyAttackbleLayers so you'll only bash windows on harm intent
  * Create a new TileInteraction SO which performs the tapping validation / logic and add it to the Window tile asset's tileInteractions field.

There shouldn't be any other changes than that (except the ones related to sound manager)

Bod9001 committed:
Reverting changes

Bod9001 committed:
window knocking done the right way

chairbender commented on UnityProject/Assets/Scripts/Tilemaps/Behaviours/TilemapDamage.cs:
```diff
@@ -24,12 +24,12 @@ public class TilemapDamage : MonoBehaviour, IFireExposable
 
```
Are there any changes in this file other than formatting changes? Looks like only formatting changes, which is fine.

Bod9001 replied:
yeah, Probably is because I copied it over from the raw on github to reverse the changes I did

chairbender committed:
Fix window knock interaction references that were showing as missing for
some reason. Only renamed the interaction to help resolve this issue, also
changed the documentation for it to match what it does.

chairbender committed:
Add interaction message for window tapping so you know who did it.

chairbender reviewed (changes requested):
Wow didn't take you long to figure out and fix. Had an issue with missing script error probably due to the naming of the window knock, also added an interaction message for it. See my PR: https://github.com/Bod9001/unitystation/pull/4

Once that's merged it's good to merge.

Bod9001 committed:
Merge pull request #4 from chairbender/Window-knocking

Fix window knock reference issues and add interaction message
